Failure to compile FFI-readline example on nix.

Hi. I get this error when I try to build the FFI-readline example.

```
[I] lc@aiur ~/S/g/i/I/s/FFI-readline (master) [127]> nix-shell -p readline --command 'idris2 --build readline.ipkg'
make: Entering directory '/home/lc/Source/github.com/idris-lang/Idris2/samples/FFI-readline/readline_glue'
gcc -Wall  -fPIC   -c -o idris_readline.o idris_readline.c
In file included from /nix/store/ahsfc5f18l7y25ykp10cp6wfqcp5qwvq-readline-6.3p08-dev/include/readline/readline.h:35,
                 from idris_readline.c:1:
/nix/store/ahsfc5f18l7y25ykp10cp6wfqcp5qwvq-readline-6.3p08-dev/include/readline/rltypedefs.h:71:28: error: unknown type name ‘FILE’
   71 | typedef int rl_getc_func_t PARAMS((FILE *));
      |                            ^~~~~~
In file included from /nix/store/ahsfc5f18l7y25ykp10cp6wfqcp5qwvq-readline-6.3p08-dev/include/readline/readline.h:36,
                 from idris_readline.c:1:
/nix/store/ahsfc5f18l7y25ykp10cp6wfqcp5qwvq-readline-6.3p08-dev/include/readline/rltypedefs.h:1:1: note: ‘FILE’ is defined in header ‘<stdio.h>’; did you forget to ‘#include <stdio.h>’?
  +++ |+#include <stdio.h>
```

--- a/samples/FFI-readline/readline_glue/idris_readline.c
+++ b/samples/FFI-readline/readline_glue/idris_readline.c
@@ -1,39 +1,30 @@
-#include <stdio.h>
 #include <readline/readline.h>
-#include <string.h>
+#include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
 
-rl_compentry_func_t* my_compentry;
+rl_compentry_func_t *my_compentry;
 
-char* compentry_wrapper(const char* text, int i) {
-    char* res = my_compentry(text, i); // Idris frees this
-    if (res != NULL) {
-        char* comp = malloc(strlen(res)+1); // Readline frees this
-        strcpy(comp, res);
-        return comp;
-    }
-    else {
-        return NULL;
-    }
-}
-
-void idrisrl_setCompletion(rl_compentry_func_t* fn) {
-    rl_completion_entry_function = compentry_wrapper;
-    my_compentry = fn;
-}
-
-char* getString(void* str) {
-    return (char*)str;
-}
-
-void* mkString(char* str) {
-    return (void*)str;
-}
-
-void* nullString() {
+char *compentry_wrapper(const char *text, int i) {
+  char *res = my_compentry(text, i); // Idris frees this
+  if (res != NULL) {
+    char *comp = malloc(strlen(res) + 1); // Readline frees this
+    strcpy(comp, res);
+    return comp;
+  } else {
     return NULL;
+  }
 }
 
-int isNullString(void* str) {
-    return str == NULL;
+void idrisrl_setCompletion(rl_compentry_func_t *fn) {
+  rl_completion_entry_function = compentry_wrapper;
+  my_compentry = fn;
 }
+
+char *getString(void *str) { return (char *)str; }
+
+void *mkString(char *str) { return (void *)str; }
+
+void *nullString() { return NULL; }
+
+int isNullString(void *str) { return str == NULL; }

--- a/samples/FFI-readline/readline_glue/idris_readline.c
+++ b/samples/FFI-readline/readline_glue/idris_readline.c
@@ -1,3 +1,4 @@
+#include <stdio.h>
 #include <readline/readline.h>
 #include <string.h>
 #include <stdlib.h>

